discovery_child_development/utils/cluster_analysis_utils.py

```diff
@@ -320,3 +321,116 @@ def cluster_keywords(
         top_cluster_tokens[unique_cluster_labels[i]] = [id_to_token[j] for j in x]
 
     return top_cluster_tokens
+
+
+def get_cluster_centroids(df: pd.DataFrame, embeddings: np.ndarray) -> List[np.ndarray]:
```

Comment:
Would this function and the next one (`get_n_most_central_vectors()`) be used chiefly with kmeans clusters or could you use them with HDBSCAN clusters too? For HDBSCAN clusters, instead of finding a centroid, would you find the point with the most neighbours?

Reply:
That's a good question. When we're looking for cluster centroids, we are assuming that the clusters are spherical. This will be the case for kmeans but might not be the case for other clustering algos like HDBSCAN. Nonetheless, I think it'll be okay to use this assumption also for HDBSCAN in most cases. 

In the end it comes down to what kind of points you think of as being "characteristic to the cluster".